packages/vite/src/node/server/sourcemap.ts

```diff
@@ -5,16 +5,26 @@ export async function injectSourcesContent(
   map: { sources: string[]; sourcesContent?: string[]; sourceRoot?: string },
   file: string
 ) {
-  const sourceRoot = await fs.realpath(
-    path.resolve(path.dirname(file), map.sourceRoot || '')
-  )
+  try {
+    var sourceRoot = await fs.realpath(
+      path.resolve(path.dirname(file), map.sourceRoot || '')
+    )
+  } catch (e) {
+    if (e.code == 'ENOENT') return
+    throw e
+  }
   map.sourcesContent = []
   await Promise.all(
     map.sources.map(async (sourcePath, i) => {
-      map.sourcesContent![i] = await fs.readFile(
-        path.resolve(sourceRoot, decodeURI(sourcePath)),
-        'utf-8'
-      )
+      try {
+        map.sourcesContent![i] = await fs.readFile(
+          path.resolve(sourceRoot, decodeURI(sourcePath)),
```

Comment:
I think we don't need to catch `ENOENT` errors here, since it's not possible for a virtual module to be in the `sources` of a real file's sourcemap. Therefore, we can know that failed `fs.readFile` calls are related to a malformed source path.